src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1007,7 +1017,20 @@ static inline bool notification_match(reservation_t& res,
         return ret;
       }
     }
-    res.topics.emplace_back(topic_filter.s3_id, topic_cfg, res_id);
+    // load the topic,if there is change in topic config while it's stored in
+    // notification.
+    rgw_pubsub_topic result;
+    const RGWPubSub ps(res.store, res.user_tenant);
+    auto ret = ps.get_topic(res.dpp, topic_cfg.name, result, res.yield);
+    if (ret < 0) {
+      ldpp_dout(res.dpp, 1)
+          << "ERROR: failed to load topic: " << topic_cfg.name
+          << ". error: " << ret
+          << ", using the stored topic from bucket notification" << dendl;
```

Comment:
do we store the topic conf in the notification also in v2?

Reply:
> do we store the topic conf in the notification also in v2?

Yea we still do continue to store it.

Reply:
i don't have a strong opinion here either way, but what is the usecase that you expect to get the topic conf from the notification and not from the topic?

Reply:
> i don't have a strong opinion here either way, but what is the usecase that you expect to get the topic conf from the notification and not from the topic?

i am using that conf in lot of places for non-notification purpose.